app/models/location.rb

```diff
@@ -593,25 +593,25 @@ def self.countries_by_count
     CountryCounter.new.countries_by_count
   end
 
-  @@location_cache = nil
+  def self.location_name_cache
```

Comment:
Projects have Locations. Do those need to be taken into account?

Reply:
Good question. For now, if the create project form allows people to sneak out without defining the location, yes, we should search for undefined strings in Projects. It's a tiny query too. 

With the new Location UI though, I'm trying to channel people to easily create locations in any context, and eventually seal off the exits so they cannot litter the db with undefined location strings that an admin must then deal with. When this is done, we can maybe delete all this code (here and elsewhere) that looks for stray "location" strings. 

But I haven't implemented the new Location UI for either `Lists` or `Projects` yet, nor have i disabled the ability to create objects with undefined "locations". 

So, "yes".

Reply:
Based on manual testing, you can create a Project without a Location.
@mo-nathan: Seems that we shouldn't allow that.

Reply:
There are expected to be Projects without Locations.  The vast majority (193 vs. 11) of existing Projects do not have a Location.  Projects for forays with locations are only one use case.

Reply:
Oops. Completely forgot that obvious fact; I'm too focused on Field Slips.

Reply:
Update: Projects do not have a `where` column, or any column for location strings. (Whew!) 

No need to check for orphaned strings there.